test/extended/builds/pipeline.go

```diff
@@ -155,6 +156,7 @@ var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
 			// start the build
 			go func() {
 				defer g.GinkgoRecover()
+				defer wg.Done()
```

Comment:
Yes, your understanding is correct.  In addition to just being cleaner, if my recollection serves, I've seen it mentioned in other PRs that doing the `defer wg.Done` asap is a best practice in this space.